### DISCUSSION TITLE
Is there a way to include KaTeX extensions? (mhchem extension)

### DISCUSSION THREAD

**Discussion by vedranmiletic** [marp-team/marp#99] [Q&A] [answered]
I looked into `katexOption` but I don't see how would I use it for [adding mhcem](https://katex.org/docs/libs.html#extensions). It requires an inclusion of extra script:

```html
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.13.3/dist/contrib/mhchem.min.js" integrity="sha384-LIgAiYlGSAdpNC9+YDjDPF6JeS/RRIumtNo0CmyQERZ/+g0h9MbuYQwf/5pQ4Y4M"  crossorigin="anonymous"></script>
```

I also tried using MathJax, but it also [requires specifying mhchem as an extension](https://docs.mathjax.org/en/latest/input/tex/extensions/mhchem.html).

**Comment by yhatt** [comment, marked as answer]
Marp's markdown rendering will be completed within Node.js. `<script>` is for browser side so will never work. Instead, you should follow KaTeX documentation for Node.js: https://github.com/KaTeX/KaTeX/blob/master/docs/node.md#using-mhchem-extension

In Marp CLI:

```js
// marp.config.js
const katex = require('katex');
require('katex/dist/contrib/mhchem.js'); // modify katex module
```

To modify Marp Core's KaTeX, you may be required installing into local.

```bash
npm i @marp-team/marp-core @marp-team/marp-cli
npx marp -c marp.config.js your-markdown.md
```

**Reply by Serenade600** [reply]
> Marp's markdown rendering will be completed within Node.js. `<script>` is for browser side so will never work. Instead, you should follow KaTeX documentation for Node.js: https://github.com/KaTeX/KaTeX/blob/master/docs/node.md#using-mhchem-extension
> 
> In Marp CLI:
> 
> ```js
> // marp.config.js
> const katex = require('katex');
> require('katex/dist/contrib/mhchem.js'); // modify katex module
> ```
> 
> To modify Marp Core's KaTeX, you may be required installing into local.
> 
> ```shell
> npm i @marp-team/marp-core @marp-team/marp-cli
> npx marp -c marp.config.js your-markdown.md
> ```

As a complete newbie I am, so the solution also fits to marp for VS-code, right?

**Reply by yhatt** [reply]
No, it works only in Marp CLI for now. We disallow custom configuration in VS Code due to potentially security risk: https://github.com/marp-team/marp-vscode/issues/194
